lib/generateError.js

```diff
@@ -0,0 +1,46 @@
+const _ = require('lodash'),
+  wap = require('webapi-parser').WebApiParser,
+  UserError = require('./UserError');
+
+/**
+ * Validates given RAML 1.0 definition using WebAPI Parser and
+ * generates UserError or Unhandled error depending upon type of error.
+ *
+ * @param {Object} definition - RAML definition
+ * @param {*} error - Original Error object
+ * @param {Function} cb - Callback function
+ * @returns {*} - Generated Error object
+ */
+function generateError (definition, error, cb) {
+  wap.raml10.parse(definition, '')
```

Comment:
@VShingala This seems like a roundabout way to generate error. Shouldn't the parsing and validation happen in the first step, rather than after an exception is raised?

Reply:
@webholik If we validate first and throw an error, many imports will fail immediately since most of the imported definitions have few things missing or invalid. We only want to categorize errors in case of actual failures so we're doing it after the conversion fails.